modules/internal/StringCasts.chpl

```diff
@@ -75,43 +83,60 @@ module StringCasts {
     return ret;
   }
 
-  inline proc _cast(type t, x: string) where isIntegralType(t) {
+  inline proc _cast(type t, x: string) throws where isIntegralType(t) {
     //TODO: switch to using qio's readf somehow
     pragma "insert line file info"
-    extern proc c_string_to_int8_t  (x:c_string) : int(8);
+    extern proc c_string_to_int8_t  (x:c_string, ref err: bool) : int(8);
     pragma "insert line file info"
-    extern proc c_string_to_int16_t (x:c_string) : int(16);
+    extern proc c_string_to_int16_t (x:c_string, ref err: bool) : int(16);
```

Comment:
CString also has these: ```extern proc c_string_to_int16_t (x:c_string) : int(16);```
which is presumably now out of date.
Can we remove the C string ones? Either this PR won't pass testing, or they are always unused. Can you check?